staging/src/k8s.io/apiserver/pkg/admission/plugin/cel/controller.go

```diff
@@ -366,7 +366,12 @@ func (c *celAdmissionController) Validate(
 				}
 				switch decision.kind {
 				case admit:
-					celmetrics.Metrics.ObserveCheck(ctx, decision.elapsed, definition.Name, binding.Name, decision.expression, "admit", paramStr, "active")
+					if decision.message != "" {
+						// the decision message is set in the admit case if and only if
+						// an ignored error occurred, so use that to filter for ignored
+						// errors (as opposed to validations that occurred with no errors)
+						celmetrics.Metrics.ObserveCheck(ctx, decision.elapsed, definition.Name, binding.Name, decision.expression, "admit", paramStr, "active")
```

Comment:
Here is how webhook do metrics:
https://github.com/kubernetes/kubernetes/blob/1f9e20eb8617e312f9f61d28d93ed29e988d953d/staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/validating/dispatcher.go#L150-L172

Could we do similar and have ObservePolicyRejection? Also I am ok with adding metrics for all admit path as long as the metric is clear

Reply:
I am ok with either track all admitted or track err but ignored. I feel it would definitely brings value to the user with those been differentiated

Reply:
For now I think we could go with tracking ignored errors (there's no easy way to make a distinction without changing the metrics a lot), and I think ignored errors would probably be more valuable to the user to track. I've also replaced `ObserveCheck` with `ObserveAdmission` and `ObserveRejection`, and the documentation for `ObserveAdmission` should hopefully make it clear that it's meant for ignored errors.